spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/result/view/FluxWriter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.web.reactive.result.view;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.function.Supplier;
+
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.io.buffer.DataBuffer;
+
+/**
+ * A {@link Writer} that can write a {@link Flux} (or {@link Publisher}) to a data buffer.
+ * Used to render progressive output in a {@link MustacheView}.
+ *
+ * @author Dave Syer
+ */
+class FluxWriter extends Writer {
+
+	private final Supplier<DataBuffer> factory;
+
+	private final Charset charset;
+
+	private Flux<String> buffers;
```

Comment:
Why can't this be `List<String>` which at the end in `getBuffers` becomes `Flux.fromIterable(this.buffers).map(...)`?

Reply:
I don’t think that would work, would it? How would you `concatMap` another `Publisher` onto it?

Reply:
Oh I see the `write(Object)` method which handles Publisher values. Still isn't it all collected/aggregated before `response#writeAndFlushWith` is called? So it could be a `List<Object>` (either String or Publisher) which can then be handled through a combination of `Flux.fromIterable` and `concatWith`.

Reply:
If there’s an advantage to doing it that way, I suppose so. What would be the point - you’d end up having to iterate over the list and do the same at the end anyway, wouldn’t you?

Reply:
For efficiency I guess, accumulating vs composing with a new operator for every write. If there are a lot of writes, this could end up with a lot of unecessary objects and an unnecessarily long pipeline.

Reply:
Doesn’t reactor optimize that away? Could it?

Reply:
Here's a list-based refactoring: 35a609d. Is that what you had in mind?

Reply:
Here is a further simplification https://github.com/rstoyanchev/spring-boot/commit/fc5a2cb89243e1cf109c22cb6dae7482e692cf60.